src/Codeception/Test/Loader/Cest.php

```diff
@@ -50,6 +52,27 @@ function ($v) {
                         },
                         $rawExamples
                     );
+                }
+
+                // dataprovider Annotation
+                $dataMethod = Annotation::forMethod($testClass, $method)->fetch('dataprovider');
+                if(!empty($dataMethod)) {
+                    if (false === is_callable([$unit, $dataMethod])) {
+                        throw new TestParseException(
+                            $file, "DataProvider $dataMethod for $testClass->$method is invalid or not callable."
+                            . PHP_EOL .
+                            "Make sure that the dataprovider method is a public static function."
+                        );
+                    }
+                    // allow to mix axample and dataprovider annotations
+                    if(count($examples)) {
+                        $examples = array_merge($examples, $testClass::$dataMethod());
+                    } else {
+                        $examples = $testClass::$dataMethod();
```

Comment:
static call can be replaced with `call_user_func([$unit, $dataMethod])` to call dataprovider method.
Also dataprovider can be allowed to be marked as protected, so it won't need the `_` prefix in it. This way we can use `ReflectionHelper::invokePrivateMethod()` to call it. 

I'd suggest to have data providers as protected methods. It should be mentioned in documentations that they at least should be protected or starting with `_` prefix so they won't become tests by themselves